lib/transport.js

```diff
@@ -63,7 +63,11 @@ const createUdpTransport = args => {
         }
         dnsResolutionData.resolvedAddress = address;
         dnsResolutionData.timestamp = now;
-        socket.send(buf, 0, buf.length, args.port, dnsResolutionData.resolvedAddress, callback);
+        try {
+          socket.send(buf, 0, buf.length, args.port, dnsResolutionData.resolvedAddress, callback);
+        } catch (socketError) {
+          callback(socketError);
+        }
       });
     } else {
       socket.send(buf, 0, buf.length, args.port, dnsResolutionData.resolvedAddress, callback);
```

Comment:
Could you change other cases of send so all codpaths now all works the same?

```suggestion
        try {
          socket.send(buf, 0, buf.length, args.port, dnsResolutionData.resolvedAddress, callback);
        } catch (socketError) {
          callback(socketError);
        }
```

Reply:
There's actually one more below this that I wanted to suggest as well, but GitHub won't let me add in a suggestion for it.

Reply:
This branch is ok because the exception thrown here is handled at the call site: https://github.com/brightcove/hot-shots/blob/master/lib/statsd.js#L380-L385. The other branch is problematic because the sync exception is thrown in another callback. That said, it looks like wrapping this will result in the same exact behavior, so I'm happy wrap the other socket.sends if you prefer that for consistency.

Reply:
Thanks, I missed that here.  If you could wrote the other socket I'd appreciate it.